src/transformer.ts

```diff
@@ -2,12 +2,27 @@ import Tag from './tag';
 import { Class } from './schema-types/class';
 import { isPromise } from './utils';
 import type { Config, Node, NodeType, Schema, Transformer } from './types';
+import { Value } from '..';
 
 type AttributesSchema = Schema['attributes'];
 
 export const globalAttributes: AttributesSchema = {
   class: { type: Class, render: true },
-  id: { type: String, render: true },
+  id: {
+    type: String,
+    render: true,
+    validate(value: Value, _config: Config) {
+      return typeof value === 'string' && value.match(/^[a-zA-Z]/)
```

Comment:
@rpaul-stripe is this true of all identifiers? Should we update `isIdentifier` so that they can't start with a number?

Reply:
No, this is enforced only on the string value of the HTML `id` attribute, not all the things that are "identifiers" in the Markdoc tag grammar.

In the current implementation (before this PR) the validator already has a [hard-coded rule for ids](https://github.com/markdoc/markdoc/blob/main/src/validator.ts#L193) that prohibits starting them with a number. I went further in this PR and made it require a letter at the beginning in order to prevent people from making mistakes like e.g. `id="#foo"`. This is also consistent with the guidance on MDN: https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/id#sect2

Granted, we may want to consider enforcing this rule for all Markdoc identifiers—but that would be a backwards-incompatible change since the grammar has always been permissive about this.

Reply:
This makes sense, thanks!